specs/latest/2.0/index.html

```diff
@@ -1092,6 +1094,52 @@ <h2 class="no-toc">Table of contents</h2>
         it is the responsibility of the WebGL API to ensure that data are accessed consistently. This applies
         even if the buffer is currently bound to a transform feedback binding point.
       </dd>
+      <dt>
+        <p class="idl-code">Promise&lt;ArrayBuffer&gt; getBufferSubDataAsync(GLenum target, GLintptr offset, ArrayBuffer returnedData)
+        </p>
+      </dt>
+      <dd>
+        For the WebGLBuffer bound to the passed target, read <code>returnedData.byteLength</code>
+        bytes from the buffer starting at byte offset <code>offset</code> and write them
+        to <code>returnedData</code>. This command executes serially with respect to others in the
+        command stream, but the data is only written to <code>returnedData</code> when the promise
+        is fulfilled.
+
+        <br><br>
+        If any of the following error conditions occur:
+
+        <ul>
+          <li><code>offset + returnedData.byteLength</code> would extend beyond the end of the buffer</li>
+          <li><code>offset</code> is less than zero</li>
+          <li>no WebGLBuffer is bound to <code>target</code></li>
+          <li><code>target</code> is <code>TRANSFORM_FEEDBACK_BUFFER</code>, and any transform feedback object
+            is currently active</li>
+        <ul>
+
+        <br><br>
+
+        then the returned Promise is rejected with an "InvalidStateError" DOMException, and no data
+        is written to <code>returnedData</code>.
+
+        <br><br>
+
+        Otherwise, the Promise is fulfilled when the data becomes available,
```

Comment:
I think there possible asynchronous failures that need to be covered here, which should probably result in the promise being rejected (timeout, context lost)?